tests/test_financial.py

```diff
@@ -90,13 +90,27 @@ def test_decimal_with_when(self):
 
 
 class TestPV:
-    def test_pv(self):
-        assert_almost_equal(npf.pv(0.07, 20, 12000, 0), -127128.17, 2)
-
-    def test_pv_decimal(self):
-        assert_equal(npf.pv(Decimal('0.07'), Decimal('20'), Decimal('12000'),
-                            Decimal('0')),
-                     Decimal('-127128.1709461939327295222005'))
+    @pytest.mark.parametrize("rate, periods, payments, future_value, exp_result", [
+                             (0.07, 20, 0, 12000, -3101.0280337664),
+                             (0.025, 4, 0, 1000, -905.9506447998),
+                             (-0.07, 56, 0, 1200000, -69845129.5182595000),
+                             (0.05, 20, 0, -12000, 4522.6737944760),
+                             (0.05, 356.5288568, 0, -25000.00, 0.0006971776),
+                             (0.05, 50, 0, 0, 0.0000000000),
+                             (0.05, 26, 500, 0, -7187.5926504975),
+                             ])
```

Comment:
These values should be strings, not floats. A float stores an inexact representation of most numbers, the decimal representation of that number will not be the same as the floating point representation. For example see:

```python
>>> from decimal import Decimal
>>> Decimal(0.1) == Decimal("0.1")
False
>>> Decimal(0.1)
Decimal('0.1000000000000000055511151231257827021181583404541015625')
>>> Decimal("0.1")
Decimal('0.1')
```